object.Head RPC with full-headers flag leads to panic

Affected version: `neofs-cli v0.7.0`

It seems like stringer for object header is completely broken now:

```
$ ./bin/neofs-cli --config .neofs-cli.yml.remote object head --full-headers --cid 4ZJydjUYwnrHEDPBhfNE1gmNNRPfggTjPSUXj6xgEXY1 --oid 163beb35-57f5-4758-b771-d5efd8d68a4f
System headers:
  Object ID   : 163beb35-57f5-4758-b771-d5efd8d68a4f
  Owner ID    : AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y
  Container ID: 4ZJydjUYwnrHEDPBhfNE1gmNNRPfggTjPSUXj6xgEXY1
  Payload Size: 131.701KB
  Version     : 1
  Created at  : epoch #20, 2020-04-29 17:50:10 +0300 MSK
Other headers:
panic: invalid Go type hash.Hash for field object.Header.HomoHash

goroutine 1 [running]:
google.golang.org/protobuf/internal/impl.newSingularConverter(0xe265c0, 0xc8a420, 0xe23220, 0xc00020c540, 0x1, 0x0)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/google.golang.org/protobuf/internal/impl/convert.go:143 +0xff5
google.golang.org/protobuf/internal/impl.NewConverter(0xe265c0, 0xc8a420, 0xe23220, 0xc00020c540, 0x0, 0x0)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/google.golang.org/protobuf/internal/impl/convert.go:60 +0xd3
google.golang.org/protobuf/internal/impl.fieldInfoForOneof(0xe23220, 0xc00020c540, 0xb84da3, 0x5, 0x0, 0x0, 0xe265c0, 0xc23c20, 0xb84daa, 0x16, ...)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/google.golang.org/protobuf/internal/impl/message_reflect_field.go:42 +0x1bd
google.golang.org/protobuf/internal/impl.(*MessageInfo).makeKnownFieldsFunc(0xc0004228c0, 0x28, 0xffffffffffffffff, 0x10, 0xffffffffffffffff, 0xc000518a50, 0xc000518a80, 0xc000518ab0, 0xc000518ae0)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/google.golang.org/protobuf/internal/impl/message_reflect.go:57 +0x33a
google.golang.org/protobuf/internal/impl.(*MessageInfo).makeReflectFuncs(0xc0004228c0, 0xe265c0, 0xc7c940, 0x28, 0xffffffffffffffff, 0x10, 0xffffffffffffffff, 0xc000518a50, 0xc000518a80, 0xc000518ab0, ...)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/google.golang.org/protobuf/internal/impl/message_reflect.go:36 +0x63
google.golang.org/protobuf/internal/impl.(*MessageInfo).initOnce(0xc0004228c0)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/google.golang.org/protobuf/internal/impl/message.go:91 +0x182
google.golang.org/protobuf/internal/impl.(*MessageInfo).init(0xc0004228c0)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/google.golang.org/protobuf/internal/impl/message.go:73 +0x3c
google.golang.org/protobuf/internal/impl.(*messageReflectWrapper).WhichOneof(0xc0000fa0f0, 0xe1e1c0, 0xc00023a480, 0xc00020c000, 0xc40a00)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/google.golang.org/protobuf/internal/impl/message_reflect_gen.go:233 +0x33
github.com/golang/protobuf/proto.(*textWriter).writeMessage(0xc000518810, 0xe20280, 0xc0000fa0f0, 0x0, 0x0)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/github.com/golang/protobuf/proto/text_encode.go:273 +0x96b
github.com/golang/protobuf/proto.(*TextMarshaler).marshal(0x13582c6, 0xe10560, 0xc000001980, 0x1, 0x1, 0xf, 0x0, 0x0)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/github.com/golang/protobuf/proto/text_encode.go:86 +0x18a
github.com/golang/protobuf/proto.(*TextMarshaler).Text(...)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/github.com/golang/protobuf/proto/text_encode.go:44
github.com/golang/protobuf/proto.CompactTextString(...)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/github.com/golang/protobuf/proto/text_encode.go:106
github.com/nspcc-dev/neofs-api-go/object.(*Header).String(...)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/github.com/nspcc-dev/neofs-api-go/object/types.pb.go:231
main.head(0xc00031a500, 0x5, 0x8)
        /home/alexvanin/Project/NeoFS/neofs-cli/object.go:261 +0x9e1
github.com/urfave/cli/v2.(*Command).Run(0xc00013f200, 0xc000167080, 0x0, 0x0)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/github.com/urfave/cli/v2/command.go:161 +0x4e0
github.com/urfave/cli/v2.(*App).RunAsSubcommand(0xc000682480, 0xc000166d80, 0x0, 0x0)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/github.com/urfave/cli/v2/app.go:433 +0xa9a
github.com/urfave/cli/v2.(*Command).startApp(0xc00013e5a0, 0xc000166d80, 0x7fffde63c067, 0x6)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/github.com/urfave/cli/v2/command.go:275 +0x67f
github.com/urfave/cli/v2.(*Command).Run(0xc00013e5a0, 0xc000166d80, 0x0, 0x0)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/github.com/urfave/cli/v2/command.go:91 +0xa12
github.com/urfave/cli/v2.(*App).RunContext(0xc000682300, 0xe14b60, 0xc0001302e8, 0xc000138000, 0xa, 0xa, 0x0, 0x0)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/github.com/urfave/cli/v2/app.go:302 +0x814
github.com/urfave/cli/v2.(*App).Run(...)
        /home/alexvanin/Project/NeoFS/neofs-cli/vendor/github.com/urfave/cli/v2/app.go:211
main.main()
        /home/alexvanin/Project/NeoFS/neofs-cli/main.go:22 +0x25c
```

New object stringer will be available with later `neofs-api-go` release (see https://github.com/nspcc-dev/neofs-api-go/pull/68), but current `neofs-api-go` and `neofs-node` versions are LTS, so we need to fix this.

--- a/README.md
+++ b/README.md
@@ -165,19 +165,38 @@ $ ./bin/neofs-cli --host fs.nspcc.ru:8080 --key ./key object head \
 --oid e35f3596-2cde-4d3e-b57a-752ed687b79a \
 --full-headers
 
-System headers:
-  Object ID   : e35f3596-2cde-4d3e-b57a-752ed687b79a
-  Owner ID    : AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y
-  Container ID: 7Gi7c1WmyKxEW3JwqEETupNoQ7rAb1CSQYxdPirXLwaG
-  Payload Size: 3.1M
-  Version     : 1
-  Created at  : epoch #13, 2019-11-07 17:42:57 +0300 MSK
-Other headers:
-  UserHeader:<Key:"Nicename" Value:"cat_picture.png" > 
-  Verify:<PublicKey:"\002{...}" >
-  HomoHash:... 
-  PayloadChecksum:"..."
-  Integrity:<HeadersChecksum:"..." ChecksumSignature:"..." >
+Object:
+	SystemHeader:
+		- ID=7e0b9c6c-aabc-4985-949e-2680e577b48b
+		- CID=11111111111111111111111111111111
+		- OwnerID=ALYeYC41emF6MrmUMc4a8obEPdgFhq9ran
+		- Version=1
+		- PayloadLength=1
+		- CreatedAt={UnixTime=1 Epoch=1}
+	ExtendedHeaders:
+		- Type=Link
+		  Value={Type=Child ID=7e0b9c6c-aabc-4985-949e-2680e577b48b}
+		- Type=Redirect
+		  Value={CID=11111111111111111111111111111111 OID=7e0b9c6c-aabc-4985-949e-2680e577b48b}
+		- Type=UserHeader
+		  Value={Key=test_key Val=test_value}
+		- Type=Transform
+		  Value=Split
+		- Type=Tombstone
+		  Value=MARKED
+		- Type=Verify
+		  Value={PublicKey=010203040506 Signature=010203040506}
+		- Type=HomoHash
+		  Value=00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
+		- Type=PayloadChecksum
+		  Value=010203040506
+		- Type=Integrity
+		  Value={Checksum=010203040506 Signature=010203040506}
+		- Type=StorageGroup
+		  Value={DataSize=5 Hash=00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 Lifetime={Unit=UnixTime Value=555}}
+		- Type=PublicKey
+		  Value=010203040506
+	Payload: []byte{0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7}
 ```
 
 You can also search by well known or user defined headers
@@ -224,21 +243,47 @@ $ ./bin/neofs-cli --host fs.nspcc.ru:8080 --key ./key sg get \
 --cid 7Gi7c1WmyKxEW3JwqEETupNoQ7rAb1CSQYxdPirXLwaG \
 --sgid a220d19f-78ca-4574-ac1b-d7b246e929b5
 
-System headers:
-  Object ID   : a220d19f-78ca-4574-ac1b-d7b246e929b5
-  Owner ID    : AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y
-  Container ID: 7Gi7c1WmyKxEW3JwqEETupNoQ7rAb1CSQYxdPirXLwaG
-  Payload Size: 0B
-  Version     : 1
-  Created at  : epoch #16, 2019-11-07 17:59:57 +0300 MSK
-Other headers:
-  Link:<type:StorageGroup ID:e35f3596-2cde-4d3e-b57a-752ed687b79a > 
-  Link:<type:StorageGroup ID:79ecc573-92c9-4066-8546-96e16e980700 > 
-  StorageGroup:<ValidationDataSize:6500758 ValidationHash:... > 
-  Verify: ...
-  HomoHash: ... 
-  PayloadChecksum: ...
-  Integrity: ...
+Object:
+	SystemHeader:
+		- ID=7e0b9c6c-aabc-4985-949e-2680e577b48b
+		- CID=11111111111111111111111111111111
+		- OwnerID=ALYeYC41emF6MrmUMc4a8obEPdgFhq9ran
+		- Version=1
+		- PayloadLength=1
+		- CreatedAt={UnixTime=1 Epoch=1}
+	ExtendedHeaders:
+		- Type=Link
+		  Value={Type=Child ID=7e0b9c6c-aabc-4985-949e-2680e577b48b}
+		- Type=Redirect
+		  Value={CID=11111111111111111111111111111111 OID=7e0b9c6c-aabc-4985-949e-2680e577b48b}
+		- Type=UserHeader
+		  Value={Key=test_key Val=test_value}
+		- Type=Transform
+		  Value=Split
+		- Type=Tombstone
+		  Value=MARKED
+		- Type=Verify
+		  Value={PublicKey=010203040506 Signature=010203040506}
+		- Type=HomoHash
+		  Value=00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
+		- Type=PayloadChecksum
+		  Value=010203040506
+		- Type=Integrity
+		  Value={Checksum=010203040506 Signature=010203040506}
+		- Type=StorageGroup
+		  Value={DataSize=5 Hash=00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 Lifetime={Unit=UnixTime Value=555}}
+		- Type=PublicKey
+		  Value=010203040506
+	Payload: []byte{0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7}
+```
+
+In the same way you can remove StorageGroup:
+
+```
+$ ./bin/neofs-cli --host fs.nspcc.ru:8080 --key ./key sg delete \
+--cid 7Gi7c1WmyKxEW3JwqEETupNoQ7rAb1CSQYxdPirXLwaG \
+--sgid a220d19f-78ca-4574-ac1b-d7b246e929b5
+
 ```
 
 ### Status operations
@@ -340,6 +385,18 @@ $ ./bin/neofs-cli --host fs.nspcc.ru:8080 --key /key status dump_vars --beauty
 {
   // ...
 }
+```
+
+#### Change state of specified node:
+
+```
+// --state can be `online` or `offline`
+
+$ ./bin/neofs-cli --host fs.nspcc.ru:8080 --key /key status change_state --state offline
+
+DONE
+
+  
 ```
 
 ## License

--- a/object.go
+++ b/object.go
@@ -253,11 +253,6 @@ func head(c *cli.Context) error {
 
 // objectStringify converts object into string format.
 func objectStringify(dst io.Writer, obj *object.Object) error {
-	// put empty line
-	if _, err := fmt.Fprintln(dst); err != nil {
-		return err
-	}
-
 	// put object line
 	if _, err := fmt.Fprintln(dst, "Object:"); err != nil {
 		return err
@@ -293,7 +288,7 @@ func objectStringify(dst io.Writer, obj *object.Object) error {
 	}
 
 	// put user headers
-	if _, err := fmt.Fprintln(dst, "\tUserHeaders:"); err != nil {
+	if _, err := fmt.Fprintln(dst, "\tExtendedHeaders:"); err != nil {
 		return err
 	}
 
@@ -342,7 +337,7 @@ func objectStringify(dst io.Writer, obj *object.Object) error {
 					buf.WriteString(" Lifetime={")
 					buf.WriteString("Unit=" + lt.Unit.String())
 					buf.WriteString(" Value=" + strconv.FormatInt(lt.Value, 10))
-					buf.WriteString(" }")
+					buf.WriteByte('}')
 				}
 				buf.WriteByte('}')
 				val = buf.String()

--- a/object.go
+++ b/object.go
@@ -8,9 +8,9 @@ import (
 	"io"
 	"math"
 	"os"
+	"reflect"
 	"strconv"
 	"strings"
-	"time"
 
 	"github.com/nspcc-dev/neofs-api-go/hash"
 	"github.com/nspcc-dev/neofs-api-go/object"
@@ -248,33 +248,127 @@ func head(c *cli.Context) error {
 		return errors.Wrap(err, "can't perform HEAD request")
 	}
 
-	fmt.Println("System headers:")
-	fmt.Printf("  Object ID   : %s\n", resp.Object.SystemHeader.ID)
-	fmt.Printf("  Owner ID    : %s\n", resp.Object.SystemHeader.OwnerID)
-	fmt.Printf("  Container ID: %s\n", resp.Object.SystemHeader.CID)
-	fmt.Printf("  Payload Size: %s\n", object.ByteSize(resp.Object.SystemHeader.PayloadLength))
-	fmt.Printf("  Version     : %d\n", resp.Object.SystemHeader.Version)
-	fmt.Printf("  Created at  : epoch #%d, %s\n", resp.Object.SystemHeader.CreatedAt.Epoch, time.Unix(resp.Object.SystemHeader.CreatedAt.UnixTime, 0))
-	if len(resp.Object.Headers) != 0 {
-		fmt.Println("Other headers:")
-		for i := range resp.Object.Headers {
-			// todo replace all this with object.Stringify in future
-			switch hdr := resp.Object.Headers[i].Value.(type) {
-			case *object.Header_HomoHash:
-				fmt.Printf("  homohash = %s\n", hdr.HomoHash.String())
-			case *object.Header_PublicKey:
-				fmt.Printf("  pubkey = %02x\n", hdr.PublicKey.Value)
-			case *object.Header_PayloadChecksum:
-				fmt.Printf("  payload_checksum = %02x\n", hdr.PayloadChecksum)
-			case *object.Header_Integrity:
-				fmt.Printf("  integrity = %s\n", hdr.Integrity.String())
-			case *object.Header_Verify:
-				fmt.Printf("  pubkey = %02x, signature = %02x\n", hdr.Verify.PublicKey, hdr.Verify.KeySignature)
-			case *object.Header_UserHeader:
-				fmt.Printf("  %s = %s\n", hdr.UserHeader.Key, hdr.UserHeader.Value)
-			default:
-				fmt.Printf("  %#v\n", resp.Object.Headers[i])
+	return objectStringify(os.Stdout, resp.Object)
+}
+
+// objectStringify converts object into string format.
+func objectStringify(dst io.Writer, obj *object.Object) error {
+	// put empty line
+	if _, err := fmt.Fprintln(dst); err != nil {
+		return err
+	}
+
+	// put object line
+	if _, err := fmt.Fprintln(dst, "Object:"); err != nil {
+		return err
+	}
+
+	// put system headers
+	if _, err := fmt.Fprintln(dst, "\tSystemHeader:"); err != nil {
+		return err
+	}
+
+	sysHeaders := []string{"ID", "CID", "OwnerID", "Version", "PayloadLength", "CreatedAt"}
+	v := reflect.ValueOf(obj.SystemHeader)
+	for _, key := range sysHeaders {
+		var val interface{}
+
+		switch key {
+		case "CreatedAt":
+			val = fmt.Sprintf(`{UnixTime=%d Epoch=%d}`,
+				obj.SystemHeader.CreatedAt.UnixTime,
+				obj.SystemHeader.CreatedAt.Epoch)
+		default:
+			if !v.FieldByName(key).IsValid() {
+				return errors.Errorf("invalid system header key: %q", key)
 			}
+
+			val = v.FieldByName(key).Interface()
+
+		}
+
+		if _, err := fmt.Fprintf(dst, "\t\t- %s=%v\n", key, val); err != nil {
+			return err
+		}
+	}
+
+	// put user headers
+	if _, err := fmt.Fprintln(dst, "\tUserHeaders:"); err != nil {
+		return err
+	}
+
+	for _, header := range obj.Headers {
+		var (
+			typ = reflect.ValueOf(header.Value)
+			key string
+			val interface{}
+		)
+
+		switch t := typ.Interface().(type) {
+		case *object.Header_Link:
+			key = "Link"
+			val = fmt.Sprintf(`{Type=%s ID=%s}`, t.Link.Type, t.Link.ID)
+		case *object.Header_Redirect:
+			key = "Redirect"
+			val = fmt.Sprintf(`{CID=%s OID=%s}`, t.Redirect.CID, t.Redirect.ObjectID)
+		case *object.Header_UserHeader:
+			key = "UserHeader"
+			val = fmt.Sprintf(`{Key=%s Val=%s}`, t.UserHeader.Key, t.UserHeader.Value)
+		case *object.Header_Transform:
+			key = "Transform"
+			val = t.Transform.Type.String()
+		case *object.Header_Tombstone:
+			key = "Tombstone"
+			val = "MARKED"
+		case *object.Header_HomoHash:
+			key = "HomoHash"
+			val = hex.EncodeToString(t.HomoHash[:])
+		case *object.Header_PayloadChecksum:
+			key = "PayloadChecksum"
+			val = hex.EncodeToString(t.PayloadChecksum)
+		case *object.Header_Integrity:
+			key = "Integrity"
+			val = fmt.Sprintf(`{Checksum=%02x Signature=%02x}`,
+				t.Integrity.HeadersChecksum,
+				t.Integrity.ChecksumSignature)
+		case *object.Header_StorageGroup:
+			key = "StorageGroup"
+			buf := new(strings.Builder)
+			if sg := t.StorageGroup; sg != nil {
+				buf.WriteByte('{')
+				buf.WriteString("DataSize=" + strconv.FormatUint(sg.ValidationDataSize, 10))
+				buf.WriteString(" Hash=" + hex.EncodeToString(sg.ValidationHash[:]))
+				if lt := sg.Lifetime; lt != nil {
+					buf.WriteString(" Lifetime={")
+					buf.WriteString("Unit=" + lt.Unit.String())
+					buf.WriteString(" Value=" + strconv.FormatInt(lt.Value, 10))
+					buf.WriteString(" }")
+				}
+				buf.WriteByte('}')
+				val = buf.String()
+			}
+		case *object.Header_Verify:
+			key = "Verify"
+			val = fmt.Sprintf("{PublicKey=%02x Signature=%02x}",
+				t.Verify.PublicKey,
+				t.Verify.KeySignature)
+		case *object.Header_PublicKey:
+			key = "PublicKey"
+			val = hex.EncodeToString(t.PublicKey.Value)
+		default:
+			key = fmt.Sprintf("Unknown(%T)", t)
+			val = t
+		}
+
+		if _, err := fmt.Fprintf(dst, "\t\t- Type=%s\n\t\t  Value=%v\n", key, val); err != nil {
+			return err
+		}
+	}
+
+	// put payload
+	if len(obj.Payload) > 0 {
+		if _, err := fmt.Fprintf(dst, "\tPayload: %#v\n", obj.Payload); err != nil {
+			return err
 		}
 	}
 

--- a/object.go
+++ b/object.go
@@ -258,7 +258,23 @@ func head(c *cli.Context) error {
 	if len(resp.Object.Headers) != 0 {
 		fmt.Println("Other headers:")
 		for i := range resp.Object.Headers {
-			fmt.Println("  " + resp.Object.Headers[i].String())
+			// todo replace all this with object.Stringify in future
+			switch hdr := resp.Object.Headers[i].Value.(type) {
+			case *object.Header_HomoHash:
+				fmt.Printf("  homohash = %s\n", hdr.HomoHash.String())
+			case *object.Header_PublicKey:
+				fmt.Printf("  pubkey = %02x\n", hdr.PublicKey.Value)
+			case *object.Header_PayloadChecksum:
+				fmt.Printf("  payload_checksum = %02x\n", hdr.PayloadChecksum)
+			case *object.Header_Integrity:
+				fmt.Printf("  integrity = %s\n", hdr.Integrity.String())
+			case *object.Header_Verify:
+				fmt.Printf("  pubkey = %02x, signature = %02x\n", hdr.Verify.PublicKey, hdr.Verify.KeySignature)
+			case *object.Header_UserHeader:
+				fmt.Printf("  %s = %s\n", hdr.UserHeader.Key, hdr.UserHeader.Value)
+			default:
+				fmt.Printf("  %#v\n", resp.Object.Headers[i])
+			}
 		}
 	}
 

--- a/object_test.go
+++ b/object_test.go
@@ -1,0 +1,172 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/nspcc-dev/neofs-api-go/object"
+	"github.com/nspcc-dev/neofs-api-go/refs"
+	"github.com/nspcc-dev/neofs-api-go/storagegroup"
+	"github.com/nspcc-dev/neofs-crypto/test"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStringify(t *testing.T) {
+	res := `Object:
+	SystemHeader:
+		- ID=7e0b9c6c-aabc-4985-949e-2680e577b48b
+		- CID=11111111111111111111111111111111
+		- OwnerID=ALYeYC41emF6MrmUMc4a8obEPdgFhq9ran
+		- Version=1
+		- PayloadLength=1
+		- CreatedAt={UnixTime=1 Epoch=1}
+	ExtendedHeaders:
+		- Type=Link
+		  Value={Type=Child ID=7e0b9c6c-aabc-4985-949e-2680e577b48b}
+		- Type=Redirect
+		  Value={CID=11111111111111111111111111111111 OID=7e0b9c6c-aabc-4985-949e-2680e577b48b}
+		- Type=UserHeader
+		  Value={Key=test_key Val=test_value}
+		- Type=Transform
+		  Value=Split
+		- Type=Tombstone
+		  Value=MARKED
+		- Type=Verify
+		  Value={PublicKey=010203040506 Signature=010203040506}
+		- Type=HomoHash
+		  Value=00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
+		- Type=PayloadChecksum
+		  Value=010203040506
+		- Type=Integrity
+		  Value={Checksum=010203040506 Signature=010203040506}
+		- Type=StorageGroup
+		  Value={DataSize=5 Hash=00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 Lifetime={Unit=UnixTime Value=555}}
+		- Type=PublicKey
+		  Value=010203040506
+	Payload: []byte{0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7}
+`
+
+	key := test.DecodeKey(0)
+
+	uid, err := refs.NewOwnerID(&key.PublicKey)
+	require.NoError(t, err)
+
+	var oid refs.UUID
+
+	require.NoError(t, oid.Parse("7e0b9c6c-aabc-4985-949e-2680e577b48b"))
+
+	obj := &object.Object{
+		SystemHeader: object.SystemHeader{
+			Version:       1,
+			PayloadLength: 1,
+			ID:            oid,
+			OwnerID:       uid,
+			CID:           refs.CID{},
+			CreatedAt: object.CreationPoint{
+				UnixTime: 1,
+				Epoch:    1,
+			},
+		},
+		Payload: []byte{1, 2, 3, 4, 5, 6, 7},
+	}
+
+	// *Header_Link
+	obj.Headers = append(obj.Headers, object.Header{
+		Value: &object.Header_Link{
+			Link: &object.Link{ID: oid, Type: object.Link_Child},
+		},
+	})
+
+	// *Header_Redirect
+	obj.Headers = append(obj.Headers, object.Header{
+		Value: &object.Header_Redirect{
+			Redirect: &object.Address{ObjectID: oid, CID: refs.CID{}},
+		},
+	})
+
+	// *Header_UserHeader
+	obj.Headers = append(obj.Headers, object.Header{
+		Value: &object.Header_UserHeader{
+			UserHeader: &object.UserHeader{
+				Key:   "test_key",
+				Value: "test_value",
+			},
+		},
+	})
+
+	// *Header_Transform
+	obj.Headers = append(obj.Headers, object.Header{
+		Value: &object.Header_Transform{
+			Transform: &object.Transform{
+				Type: object.Transform_Split,
+			},
+		},
+	})
+
+	// *Header_Tombstone
+	obj.Headers = append(obj.Headers, object.Header{
+		Value: &object.Header_Tombstone{
+			Tombstone: &object.Tombstone{},
+		},
+	})
+
+	// *Header_Verify
+	obj.Headers = append(obj.Headers, object.Header{
+		Value: &object.Header_Verify{
+			Verify: &object.VerificationHeader{
+				PublicKey:    []byte{1, 2, 3, 4, 5, 6},
+				KeySignature: []byte{1, 2, 3, 4, 5, 6},
+			},
+		},
+	})
+
+	// *Header_HomoHash
+	obj.Headers = append(obj.Headers, object.Header{
+		Value: &object.Header_HomoHash{
+			HomoHash: object.Hash{},
+		},
+	})
+
+	// *Header_PayloadChecksum
+	obj.Headers = append(obj.Headers, object.Header{
+		Value: &object.Header_PayloadChecksum{
+			PayloadChecksum: []byte{1, 2, 3, 4, 5, 6},
+		},
+	})
+
+	// *Header_Integrity
+	obj.Headers = append(obj.Headers, object.Header{
+		Value: &object.Header_Integrity{
+			Integrity: &object.IntegrityHeader{
+				HeadersChecksum:   []byte{1, 2, 3, 4, 5, 6},
+				ChecksumSignature: []byte{1, 2, 3, 4, 5, 6},
+			},
+		},
+	})
+
+	// *Header_StorageGroup
+	obj.Headers = append(obj.Headers, object.Header{
+		Value: &object.Header_StorageGroup{
+			StorageGroup: &storagegroup.StorageGroup{
+				ValidationDataSize: 5,
+				ValidationHash:     storagegroup.Hash{},
+				Lifetime: &storagegroup.StorageGroup_Lifetime{
+					Unit:  storagegroup.StorageGroup_Lifetime_UnixTime,
+					Value: 555,
+				},
+			},
+		},
+	})
+
+	// *Header_PublicKey
+	obj.Headers = append(obj.Headers, object.Header{
+		Value: &object.Header_PublicKey{
+			PublicKey: &object.PublicKey{Value: []byte{1, 2, 3, 4, 5, 6}},
+		},
+	})
+
+	buf := new(bytes.Buffer)
+
+	require.NoError(t, objectStringify(buf, obj))
+	require.Equal(t, res, buf.String())
+}

--- a/storagegroup.go
+++ b/storagegroup.go
@@ -18,7 +18,13 @@ var (
 
 	getSGAction = &action{
 		Action: getSG,
-		Flags:  append(headObjectAction.Flags, storagegroupID),
+		Flags: []cli.Flag{
+			containerID,
+			rawQuery,
+			storagegroupID,
+			oidHidden,
+			fullHeadersHidden,
+		},
 	}
 
 	listSGAction = &action{
@@ -28,7 +34,13 @@ var (
 
 	delSGAction = &action{
 		Action: delSG,
-		Flags:  append(delObjectAction.Flags, storagegroupID),
+		Flags: []cli.Flag{
+			containerID,
+			rawQuery,
+			storagegroupID,
+			oidHidden,
+			fullHeadersHidden,
+		},
 	}
 
 	putSGAction = &action{
@@ -37,6 +49,16 @@ var (
 			containerID,
 			objectIDs,
 		},
+	}
+
+	oidHidden = &cli.StringFlag{
+		Name:   objFlag,
+		Hidden: true,
+	}
+
+	fullHeadersHidden = &cli.StringFlag{
+		Name:   fullHeadersFlag,
+		Hidden: true,
 	}
 )
 
@@ -49,7 +71,7 @@ func listSG(c *cli.Context) error {
 
 func getSG(c *cli.Context) error {
 	if err := c.Set(objFlag, c.String(sgidFlag)); err != nil {
-		return err
+		return errors.Wrap(err, "HERE!!!")
 	} else if err := c.Set(fullHeadersFlag, "true"); err != nil {
 		return err
 	}

--- a/storagegroup.go
+++ b/storagegroup.go
@@ -71,7 +71,7 @@ func listSG(c *cli.Context) error {
 
 func getSG(c *cli.Context) error {
 	if err := c.Set(objFlag, c.String(sgidFlag)); err != nil {
-		return errors.Wrap(err, "HERE!!!")
+		return err
 	} else if err := c.Set(fullHeadersFlag, "true"); err != nil {
 		return err
 	}

